### PR TITLE
Update Go plugins docs

### PR DESCRIPTION
**Pull request by buger**
My take on Go plugins improvements. 

It used tabbed content for exact Go version (btw I also fixed multiple tab blocks on one page).

Additionally, I added terms of content to this page, because it is huge. Now we have a new shortcode: `{{< toc >}}`.

https://deploy-preview-1771--tyk-docs.netlify.app/docs/nightly/plugins/supported-languages/golang/

### CODE REVIEW

**Review by marksou — APPROVED** [review]
LGTM

File: tyk-docs/content/plugins/supported-languages/golang-plugins/golang-plugins.md [hunk 1]
```diff
@@ -154,7 +188,17 @@ curl http://localhost:8181/my_api_name/get
 
 We see that the upstream target has received the header `"Foo": "Bar"` which was added by our custom middleware implemented as a native Golang plugin in Tyk.
 
-### Types of custom middleware hooks supported by Tyk Golang plugins
+#### Updating the plugin
+
+Loading an updated version of your plugin require one of the following actions:
+
+* An API reload with a NEW path or file name of your `.so` file with the plugin. You will need to update the API spec section `"custom_middleware"`, specifying a new value for the `"path"` field of the plugin you need to reload.
```
Comment (alephnull) [inline tyk-docs/content/plugins/supported-languages/golang-plugins/golang-plugins.md:195]: I have never managed to get this feature to work.

**Review by alephnull — APPROVED** [review]

**Comment by sedkis** [comment]
Tested, work perfectly.   This is a massive improvement from a DX perspective in building plugins.

One thing, the docs mention using `go vendor` but it is not required in this [sync dependencies](https://deploy-preview-1771--tyk-docs.netlify.app/docs/nightly/plugins/supported-languages/golang/#sync-dependencies) step.

File: tyk-docs/content/plugins/supported-languages/golang-plugins/golang-plugins.md [hunk 2]
```diff
@@ -55,58 +86,61 @@ We see that the Golang plugin:
...
-The plugin compiler does not support this OS as it uses glibc 2.23 which is incompatible with our standard build environment. If you absolutely must have go plugin support on Xenial, please write to us.
+A specific of Golang plugins is that they need to be built using exactly the same Tyk binary as the one to be installed. In order to make it work, we provide a special Docker image, which we internally use for building our official binaries too.
 
-### When Upgrading your Tyk Installation
-
-We release a new version of the compiler for each Tyk version. After upgrading to a new version you will need to rebuild your plugin using the new Tyk version Docker tag of the compiler.
+{{< tabs_start >}}
+{{< tab_start "v3.2.2" >}}
+```bash
+docker run --rm -v `pwd`:/plugin-source tykio/tyk-plugin-compiler:v3.2.2-rc7 plugin.so
```
Comment (sedkis) [inline tyk-docs/content/plugins/supported-languages/golang-plugins/golang-plugins.md]: can replace `v3.2.2-rc7` with `v3.2.2`

File: tyk-docs/content/plugins/supported-languages/golang-plugins/golang-plugins.md (hunk 1 above)
Comment (sedkis) [reply tyk-docs/content/plugins/supported-languages/golang-plugins/golang-plugins.md:195]: @alephnull  what do you mean?

**Pull request merged**